aten/src/ATen/native/cuda/BinaryOpsKernel.cu

```diff
@@ -1,217 +0,0 @@
-#include <ATen/Context.h>
-#include <ATen/Dispatch.h>
-#include <ATen/native/cuda/Loops.cuh>
-#include <ATen/native/DispatchStub.h>
-#include <ATen/native/TensorIterator.h>
-#include <ATen/native/BinaryOps.h>
-#include <limits>
-
-
-// NOTE: CUDA on Windows requires that the enclosing function
-// of a __device__ lambda not have internal linkage.
-
-namespace at { namespace native {
-
-void add_kernel_cuda(TensorIterator& iter, Scalar alpha_scalar) {
-  AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBool, iter.common_dtype(), "add_cuda/sub_cuda", [&]() {
-    auto alpha = alpha_scalar.to<scalar_t>();
-    gpu_kernel_with_scalars(iter, [alpha]GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-      return a + alpha * b;
-    });
-  });
-}
-
-static void sub_kernel_cuda(TensorIterator& iter, Scalar alpha_scalar) {
-  add_kernel_cuda(iter, -alpha_scalar);
-}
-
-void div_kernel_cuda(TensorIterator& iter) {
-  if (!isIntegralType(iter.common_dtype(), /*includeBool*/ false) && iter.is_cpu_scalar(2)) {
-    // optimization for floating-point types: if the second operand is a CPU
-    // scalar, compute a * reciprocal(b). Note that this may lose one bit of
-    // precision compared to computing the division.
-    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.common_dtype(), "div_cuda", [&]() {
-      auto inv_b = scalar_t(1.0 / iter.scalar_value<scalar_t>(2));
-      iter.remove_operand(2);
-      gpu_kernel(iter, [inv_b]GPU_LAMBDA(scalar_t a) -> scalar_t {
-        return a * inv_b;
-      });
-    });
-  } else {
-    AT_DISPATCH_ALL_TYPES_AND(kHalf, iter.common_dtype(), "div_cuda", [&]() {
-      gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-        return a / b;
-      });
-    });
-  }
-}
-
-void mul_kernel_cuda(TensorIterator& iter) {
-  if (iter.common_dtype() == ScalarType::Bool) {
-    // Workaround for the error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
-    gpu_kernel_with_scalars(iter, []GPU_LAMBDA(bool a, bool b) -> bool {
-      return a && b;
-    });
-  } else {
-    AT_DISPATCH_ALL_TYPES_AND(kHalf, iter.common_dtype(), "mul_cuda", [&]() {
-      gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-        return a * b;
-      });
-    });
-  }
-}
-
-void atan2_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.common_dtype(), "atan2_cuda", [&]() {
-    gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> scalar_t {
-      return ::atan2(a, b);
-    });
-  });
-}
-
-void logical_xor_kernel_cuda(TensorIterator& iter) {
-  if (iter.common_dtype() == ScalarType::Bool) {
```

Comment:
This logic is useless

Reply:
Seems like so. Perhaps deleting this logical for all comparison ops will speed up sufficiently without the need to split.

Reply:
@xuhdev Even splitted, BinaryCompareKernel.cu still takes 2min 44s to compile

Reply:
If every function in BinaryCompareKernel.cu is cut into half, then it may be reduced down to a reasonable time. I believe the functions in BinaryCompareKernel.cu might be the bottleneck.

Reply:
Hmm, this PR already cut it into half, but it still takes more than 2 minutes...